### ISSUE TITLE
Can't load or-tools 9.5 library on macOS (aarm64), .NET6  

### ISSUE THREAD

**Issue by mslazynski** [google/or-tools#3661]
Version: v9.5
Language: C#, .NET6 (6.0.405)
Solver: CP-Sat, routing
OS: macOS Ventura 13.2
CPU: Apple M1 Pro (arm64)

I'm installing or-tools 9.5 via nuget.
I expect to be able to call cp-ortools from C#. Currently, I'm a happy user of the version 9.4, which works just fine.
Instead, when using 9.5, I get a runtime exception:

``` 
Unhandled exception. System.BadImageFormatException: Could not load file or assembly 'Google.OrTools, Version=9.5.2237.0, Culture=neutral, PublicKeyToken=53dbb1ea090cbef7'. An attempt was made to load a program with an incorrect format.

File name: 'Google.OrTools, Version=9.5.2237.0, Culture=neutral, PublicKeyToken=53dbb1ea090cbef7'`
```

**Comment by mslazynski** [comment]
I've built `Google.OrTools.dll` directly from the v9.5 source tarball (provided in the github release). 
Putting it in place of the one distributed in the nuget package fixes the issue.

**Comment by lperron** [comment]
We can reproduce.

We will debug.

**Comment by sagmansercan** [comment]
I've also encountered this problem on macOS M2 arm64 apple silicon. upgrading to the latest version didn't work also. 

I've built the latest version of the project on my local and copied the .dll file into the my dotnet project build output directory and it worked. (finally..)

1. Followed the installation guide: https://developers.google.com/optimization/install/dotnet/source_mac

    at this step I've just got error from "--target ALL_BUILD" for ALL_BUILD. So, removed it :) (but not sure if this is a good idea)

2. Upgraded the package version in my project to "9.6.2534"
3. Copied "or-tools/build/dotnet/Google.OrTools/bin/arm64/Release/net6.0/osx-arm64/Google.OrTools.dll" into my project build output directory, which is myproject/bin/Debug/net6.0
